ManiVault/src/actions/DatasetPickerAction.cpp

```diff
@@ -48,35 +49,64 @@ DatasetPickerAction::DatasetPickerAction(QObject* parent, const QString& title)
         }
     });
 
+    const auto filterModelChanged = [this]() -> void {
+        if (isDatasetsChangedSignalBlocked())
+            return;
+
+        emit datasetsChanged(getDatasets());
+    };
+
+    connect(&_datasetsFilterModel, &QSortFilterProxyModel::rowsInserted, this, filterModelChanged);
+    connect(&_datasetsFilterModel, &QSortFilterProxyModel::rowsRemoved, this, filterModelChanged);
+    connect(&_datasetsFilterModel, &QSortFilterProxyModel::layoutChanged, this, filterModelChanged);
+
     populationModeChanged();
 }
 
 Datasets DatasetPickerAction::getDatasets() const
 {
-    switch (_populationMode)
-    {
-        case AbstractDatasetsModel::PopulationMode::Manual:
-            return _datasetsListModel.getDatasets();
+    Datasets datasets;
 
-        case AbstractDatasetsModel::PopulationMode::Automatic:
-            return mv::data().getDatasetsListModel().getDatasets();
+    for (std::int32_t filterModelRowIndex = 0; filterModelRowIndex < _datasetsFilterModel.rowCount(); ++filterModelRowIndex) {
+        const auto sourceModelIndex = _datasetsFilterModel.mapToSource(_datasetsFilterModel.index(filterModelRowIndex, 0));
 
-        default:
-            break;
+        if (!sourceModelIndex.isValid())
+            continue;
+
+        switch (_populationMode)
+        {
+            case AbstractDatasetsModel::PopulationMode::Manual:
+                datasets << _datasetsListModel.getDataset(sourceModelIndex.row());
+                break;
+
+            case AbstractDatasetsModel::PopulationMode::Automatic:
+                datasets << mv::data().getDatasetsListModel().getDataset(sourceModelIndex.row());
+                break;
+
+            default:
+                break;
+        }
     }
 
-    return {};
+    return datasets;
 }
 
-void DatasetPickerAction::setDatasets(Datasets datasets)
+void DatasetPickerAction::setDatasets(Datasets datasets, bool silent /*= false*/)
 {
 #ifdef DATASET_PICKER_ACTION_VERBOSE
     qDebug() << __FUNCTION__;
 #endif
 
     setPopulationMode(AbstractDatasetsModel::PopulationMode::Manual);
 
-    _datasetsListModel.setDatasets(datasets);
+    if (silent) {
+        blockDatasetsChangedSignal();
+        {
+            _datasetsListModel.setDatasets(datasets);
```

Comment:
This only sets the new datasets if `silent` is `true`, right? Shouldn't it rather always set the datasets while `silent` only affects the signal?

Reply:
You are right, that slipped through... I'll fix it

Reply:
Fixed!